docs/api.md

```diff
@@ -729,19 +728,26 @@ Shortcut for [page.mainFrame().evaluate(pageFunction, ...args)](#frameevaluatepa
 #### page.evaluateHandle(pageFunction, ...args)
 - `pageFunction` <[function]|[string]> Function to be evaluated in the page context
 - `...args` <...[Serializable]|[JSHandle]> Arguments to pass to `pageFunction`
-- returns: <[Promise]<[JSHandle]>> Resolves to the return value of `pageFunction`
+- returns: <[Promise]<[JSHandle]>> Promise which resolves to the return value of `pageFunction` as in-page object (JSHandle)
 
-If the function, passed to the `page.evaluateHandle`, returns a [Promise], then `page.evaluateHandle` would wait for the promise to resolve and return its value.
+If the function passed to the `page.evaluateHandle` returns a [Promise], then `page.evaluateHandle` would wait for the promise to resolve and return its value.
 
+The only difference between `page.evaluate` and `page.evaluateHandle` is that `page.evaluate` returns a JSON representation of the object and `page.evaluateHandle` returns in-page object (JSHandle):
 ```js
-const aWindowHandle = await page.evaluateHandle(() => Promise.resolve(window));
-aWindowHandle; // Handle for the window object.
-```
+// returns a JSON object
+const jsObject = await page.evaluate(pageFunction, elementHandle);
 
-A string can also be passed in instead of a function.
+// returns in-page object (JSHandle)
+const jsHandle = await page.evaluateHandle(pageFunction, elementHandle);
+
+// both lines below print true
+console.log(`jsObject is NOT equal to jsHandle: ${jsObject !== jsHandle}`);
```

Comment:
I'd drop this `console.log` – even two consecutive `page.evaluate` will result in strict equality failing as well:
```js
const jsObject1 = await page.evaluate(pageFunction, elementHandle);
const jsObject2 = await page.evaluate(pageFunction, elementHandle);
console.log(jsObject1 === jsObject2); // false, unless pageFunction returns primitives
```

Reply:
Good point, agree.

Reply:
Same as my comment above - it was working for me since I was using the `pageFunction` that returns `element.innterHTML`.